framework/src/modules/chain/logic/block/index.js

```diff
@@ -0,0 +1,210 @@
+const BlockOld = require('./block');
+const BlockBFT = require('./block_bft');
+
+class Block {
+	constructor(ed, schema, transaction, bftUpgradeHeight) {
+		this.scope = {
+			ed,
+			schema,
+			transaction,
+			bftUpgradeHeight,
+		};
+
+		this.blockOld = new BlockOld(ed, schema, transaction);
+		this.blockBFT = new BlockBFT(ed, schema, transaction);
+
+		this.attachSchema();
```

Comment:
All of the code in this class can be avoided by; 

```suggestion
		this.attachSchema();
		
		['create', 'sign', 'getHash', 'verifyBytes'].map(key => {
		   this[key] = (...args) => {
		      return this.blockLogicForHeight(args[0].height)[key](args); 
		   }
		});
```
```